test/k8s/datapath_configuration.go

```diff
@@ -168,38 +151,6 @@ var _ = Describe("K8sDatapathConfig", func() {
 			}
 		}
 
-		It("Check connectivity with VXLAN encapsulation", func() {
-			options := map[string]string{}
-			enableVXLANTunneling(options)
-			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
-			validateBPFTunnelMap()
-			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
-		}, 600)
-
-		// Geneve is currently not supported on GKE
-		SkipItIf(helpers.RunsOnGKE, "Check connectivity with Geneve encapsulation", func() {
-			deploymentManager.DeployCilium(map[string]string{
-				"tunnel": "geneve",
-			}, DeployCiliumOptionsAndDNS)
-			validateBPFTunnelMap()
-			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
-		})
-
-		It("Check vxlan connectivity with per-endpoint routes", func() {
-			options := map[string]string{
-				"endpointRoutes.enabled": "true",
-			}
-			enableVXLANTunneling(options)
-			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
-			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
-
-			if helpers.RunsOn419OrLaterKernel() {
-				By("Test BPF masquerade")
-				Expect(testPodHTTPToOutside(kubectl, "http://google.com", false, false, false)).
-					Should(BeTrue(), "Connectivity test to http://google.com failed")
```

Comment:
This is not covered by configuration 4 of ci-datapath. Do we know why BPF masquerading in the context of VXLAN+per endpoint routes specifically would matter?

Reply:
Good catch, but it should be covered after https://github.com/cilium/cilium/pull/24210/files has been merged.

There is nothing specific about different routing modes when it comes to BPF masq, as it's entirely handled by `bpf_host`.

Reply:
> There is nothing specific about different routing modes when it comes to BPF masq, as it's entirely handled by `bpf_host`.

I'm seeing a lot of macros on `TUNNEL_MODE` and `IS_BPF_OVERLAY` in the SNAT logic :grimacing: 
https://github.com/cilium/cilium/blob/853ec101d8adee8d482040aa038e5b71ce7775e2/bpf/lib/nat.h#L576

Reply:
Completely forgot about those. Thanks.

Anyway, #23210 was merged, so the BPF masq should be tested by ci-datapath.